stm32-modules/include/heater-shaker/heater-shaker/flash.hpp

```diff
@@ -0,0 +1,108 @@
+/**
+ * @file flash.hpp
+ * @brief Implements a FLASH class that is specialized towards
+ * holding the Thermal Offset Constants for the Heater-Shaker heat plate.
+ */
+
+#pragma once
+
+#include <cstddef>
+#include <cstdint>
+
+namespace flash {
+
+/**
+ * @brief Constant values used for calculating the offset between
+ * the physical thermistors on the system and the actual temperature on
+ * the Heater-Shaker's heat plate.
+ * @details
+ * The temperature difference between the thermistors and the surface
+ * of the Heater-Shaker tends to scale with the magnitude of the thermistor
+ * readings. Using two constants, B and C (for legacy purposes), the
+ * resulting temperature relationship can be summarized as follows:
+ *
+ * > Plate Temp = ((B + 1) * Measured Temp) + C
+ *
+ * One of the FLASH addresses is reserved for a flag to indicate whether
+ * the values have been written. The \ref FLASHFlag enum captures the
+ * valid states of this page. The page indicates what error detection,
+ * if any, is included with the FLASH constant values.
+ *
+ */
+struct OffsetConstants {
+    // The value of the constants B and C
+    double b, c, flag;
```

Comment:
this makes `flag` a `double` also, should it be a bool or something

Reply:
I made it a double for consistency and because it will eventually be converted to uint64_t to be programmed into flash

Reply:
Here in the cpp side, we should make the type be what it actually means - an enum or something. Writing it as a uint64_t is an implementation detail that should be hidden at this level.